prow/plugins/cla/cla_test.go

```diff
@@ -15,3 +15,157 @@ limitations under the License.
+			context:   "cla/linuxfoundation",
+			state:     "failure",
+			statusSHA: "a",
+			issues: []github.Issue{
+				{Number: 3, State: "open", Labels: []github.Label{{Name: claYesLabel}}},
+			},
+			pullRequests: []github.PullRequest{
+				{Number: 3, Head: github.PullRequestBranch{SHA: "b"}},
+			},
+			addedLabels:   nil,
+			removedLabels: nil,
+		},
+		{
+			name:      "cla/linuxfoundation status on head commit of PR adds the cla-yes label when its state is \"success\"",
+			context:   "cla/linuxfoundation",
+			state:     "success",
+			statusSHA: "a",
+			issues: []github.Issue{
+				{Number: 3, State: "open", Labels: []github.Label{}},
+			},
+			pullRequests: []github.PullRequest{
+				{Number: 3, Head: github.PullRequestBranch{SHA: "a"}},
+			},
+			addedLabels:   []string{fmt.Sprintf("/#3:%s", claYesLabel)},
+			removedLabels: nil,
+		},
+		{
+			name:      "cla/linuxfoundation status on head commit of PR does nothing when pending",
+			context:   "cla/linuxfoundation",
+			state:     "pending",
+			statusSHA: "a",
+			issues: []github.Issue{
+				{Number: 3, State: "open", Labels: []github.Label{}},
+			},
+			pullRequests: []github.PullRequest{
+				{Number: 3, Head: github.PullRequestBranch{SHA: "a"}},
+			},
+			addedLabels:   nil,
+			removedLabels: nil,
+		},
+		{
+			name:      "cla/linuxfoundation status success removes \"cncf-cla: no\" label",
+			context:   "cla/linuxfoundation",
+			state:     "success",
+			statusSHA: "a",
+			issues: []github.Issue{
+				{Number: 3, State: "open", Labels: []github.Label{{Name: claNoLabel}}},
+			},
+			pullRequests: []github.PullRequest{
+				{Number: 3, Head: github.PullRequestBranch{SHA: "a"}},
+			},
+			addedLabels:   []string{fmt.Sprintf("/#3:%s", claYesLabel)},
+			removedLabels: []string{fmt.Sprintf("/#3:%s", claNoLabel)},
+		},
+		{
+			name:      "cla/linuxfoundation status failure removes \"cncf-cla: yes\" label",
+			context:   "cla/linuxfoundation",
+			state:     "failure",
+			statusSHA: "a",
+			issues: []github.Issue{
+				{Number: 3, State: "open", Labels: []github.Label{{Name: claYesLabel}}},
+			},
+			pullRequests: []github.PullRequest{
+				{Number: 3, Head: github.PullRequestBranch{SHA: "a"}},
+			},
+			addedLabels:   []string{fmt.Sprintf("/#3:%s", claNoLabel)},
+			removedLabels: []string{fmt.Sprintf("/#3:%s", claYesLabel)},
+		},
+	}
+	for _, tc := range testcases {
+		pullRequests := make(map[int]*github.PullRequest)
+		for _, pr := range tc.pullRequests {
+			pullRequests[pr.Number] = &pr
+		}
+
+		fc := &fakegithub.FakeClient{
+			PullRequests: pullRequests,
+			Issues:       tc.issues,
+		}
+		se := github.StatusEvent{
+			Context: tc.context,
+			SHA:     tc.statusSHA,
+			State:   tc.state,
+		}
+		if err := handle(fc, logrus.WithField("plugin", pluginName), se); err != nil {
+			t.Errorf("For case %s, didn't expect error from cla plugin: %v", tc.name, err)
+			continue
+		}
+
+		if !reflect.DeepEqual(fc.LabelsAdded, tc.addedLabels) {
```

Comment:
I'm very wary of reflect. I think you'll only be adding/removing either 0 or 1 labels at a time, so it makes some sense to check that explicitly, then explicitly check that the label that was added/removed is the right one.

Not a big deal though.